src/ninja.cc

```diff
@@ -289,6 +293,22 @@ bool NinjaMain::RebuildManifest(const char* input_file, string* err,
   return true;
 }
 
+void NinjaMain::ParsePreviousElapsedTimes() {
+  for (vector<Edge*>::iterator edge = state_.edges_.begin(),
```

Comment:
Ninja uses C++11 now.

I strongly recommend using a range-based forloop so that this nested forloop pair isn't eye-bleed to read.